### ISSUE TITLE
Istio-Ingressgateway fails to run as non-root - Istio 1.6.3

### ISSUE THREAD

**Issue by tariq1890** [istio/istio#24954]
**Bug description**

When dropping all privileges and setting Istio-ingressgateway to run as non-root, it fails to connect to the istiod discovery endpoint and thus fails the readiness probe.  These  error logs were observed

```
2020-06-24T18:31:02.909724Z	warn	Envoy proxy is NOT ready: config not received from Pilot (is Pilot running?): cds updates: 0 successful, 0 rejected; lds updates: 0 successful, 0 rejected
2020-06-24T18:31:03.167743Z	info	sds	resource:default new connection
2020-06-24T18:31:03.167819Z	error	sds	Failed to get credential token: open ./var/run/secrets/tokens/istio-token: permission denied
2020-06-24T18:31:03.167866Z	info	sds	resource:default connection is terminated: rpc error: code = Canceled desc = context canceled
2020-06-24T18:31:03.167994Z	warning	envoy config	[bazel-out/k8-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:92] StreamSecrets gRPC config stream closed: 2, open ./var/run/secrets/tokens/istio-token: permission denied
```

[ ] Configuration Infrastructure
[ ] Docs
[X] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

Ingressgateway should connect to Istiod without issue and pass readiness probes.

**Steps to reproduce the bug**

Use the following securityContext for istio-ingressgateway

```
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          privileged: false
          readOnlyRootFilesystem: true
```

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**

**How was Istio installed?**

Istio helm charts

**Environment where bug was observed (cloud vendor, OS, etc)**

AWS - Istio 1.6.3

**Comment by tariq1890** [comment]
@howardjohn Can you help with this issue?

**Comment by howardjohn** [comment]
Looks like non root is the default on master and root is default on 1.6, so maybe there was a fix for it in master that didn't make it into 1.6?

```
$ ik manifest generate -d manifests | k grep Deployment/istio-ingressgateway | grep securityContext -A 10
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop:
            - ALL
          privileged: false
          readOnlyRootFilesystem: true
        volumeMounts:
        - mountPath: /etc/istio/proxy
          name: istio-envoy
        - mountPath: /etc/istio/config
--
      securityContext:
        fsGroup: 1337
        runAsGroup: 1337
        runAsNonRoot: true
        runAsUser: 1337
      serviceAccountName: istio-ingressgateway-service-account
      volumes:
      - configMap:
          name: istio-ca-root-cert
        name: istiod-ca-cert
      - downwardAPI:
```

^ from master

**Comment by tariq1890** [comment]
Any updates on this? We still can't get istio-ingressgateway to work as non-root

**Comment by tariq1890** [comment]
@costinm I believe you had some inputs on this. But I am not able to retrieve those messages since Istio Slack messages expire.

**Comment by tariq1890** [comment]
This can be closed as it's fixed in 1.7..x
